internal/thing/thing.go

```diff
@@ -61,6 +58,7 @@ func (t *DefaultThing) makeAuthorisedRequest(f func(session session.Session) err
 		t.session, err = builder.
 			WithConnection(t.connection).
 			AuthenticateWith(t.handlers...).
+			SignRequestsWith(t.session.(*isession.DefaultSession).SigningKey()).
```

Comment:
We shouldn't convert without checking for success as calling the `SigningKey` method would cause a runtime error if this is not a `DefaultSession` for some reason. Either `DefaultThing` should hold a `DefaultSession` or we should reinstate the `SigningKey` on the `Connection` interface.

Reply:
Good point. I think storing the key with the thing and using it here rather than getting it from the old session maintains the abstraction without exposing methods for internal use.